### PR TITLE
CI: run cilium without kube-proxy

### PR DESCRIPTION
**Pull request by cyclinder**
test macvlan work with cilium without kube-proxy.

---
name: Pull request
about: Tell us about your contribution
labels: ["pr/release/none-required"]
---

---
## Thanks for contributing !

Before submitting a pull request, make sure you read about our Contribution notice here: <https://spidernet-io.github.io/spiderpool/latest/develop/contributing/>

**What this PR does / why we need it**:

CI: run cilium without kube-proxy

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**make sure your commit is signed off**

### CODE REVIEW

File: .github/workflows/e2e-init.yaml [hunk 1]
```diff
@@ -41,6 +41,8 @@ jobs:
             e2e_test_mode: e2e_test_calico
           - e2e_init_mode: e2e_init_cilium
             e2e_test_mode: e2e_test_cilium
+          - e2e_init_mode: e2e_init_cilium_without_kube_proxy
```
Comment (weizhoublue) [inline .github/workflows/e2e-init.yaml:44]: e2e_init_underlay_with_ebpf
会更好理解是个 underlay 场景，将来更多的 ebpf 加速 也可以加入这个 target

Comment (cyclinder) [reply .github/workflows/e2e-init.yaml:44]: 那还是单独起一个target？

File: test/Makefile [hunk 2]
```diff
@@ -155,10 +157,6 @@ setup_kurise:
 
 .PHONY: setup_spiderpool
 setup_spiderpool:
-	@echo "Load Image $(E2E_SPIDERPOOL_TAG) to kind..." ; \
```
Comment (weizhoublue) [inline test/Makefile:158]: 这个可以加速 本地开发的 load 速度，尤其一些离线环境

File: test/scripts/install-multus.sh [hunk 3]
```diff
@@ -93,9 +94,16 @@ spec:
       exit 1
   esac
 
+  ENABLE_COORDINATOR=true
+  if [ "${DISABLE_KUBE_PROXY}" == "true" ] ; then
+      ENABLE_COORDINATOR=false
+      echo "DISABLE_KUBE_PROXY is true , disable coordinator config"
```
Comment (weizhoublue) [inline test/scripts/install-multus.sh:100]: 其实作为最佳实践，coordinator 还是要开 确保 宿主机联通
只要关闭 spidermultusconfig 的  spec.coordinator. podCIDRType=none   是否即可

File: test/Makefile (hunk 2 above)
Comment (cyclinder) [reply test/Makefile:158]: 下面统一load了

File: test/scripts/install-multus.sh (hunk 3 above)
Comment (cyclinder) [reply test/scripts/install-multus.sh:100]: 可

**Comment by weizhoublue** [comment]
看着 我们需要一篇 doc， 说明 cluster ip 解析的问题
（1）by kube proxy
（2）by cilium

**Comment by cyclinder** [comment]
https://github.com/spidernet-io/spiderpool/pull/2441 来做这件事吧

**Pull request merged**